src/pages/oauth/google.tsx

```diff
@@ -79,14 +86,23 @@ export const GoogleOauth2 = () => {
   const code = searchParams.get('code');
 
   useEffect(() => {
-    if (!code) {
-      window.location.href = googleLoginURL;
-      return;
+    if (Capacitor.getPlatform() === 'ios') {
+      GoogleAuth.signIn().then(googleUser => {
+        googleOauth(googleUser.authentication.idToken, referrerUser, eventId, 'ios')
+          .then(res => onLoginSucceed(res))
```

Comment:
can you make a general function for the `googleOauth` request? (because it's used in 2 places)
(I knew you used previous codes but if you could change it to `async/await`, I would greatly appreciate it)